hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java

```diff
@@ -89,6 +89,21 @@ public static boolean isJava7OrAbove() {
     return true;
   }
 
+  // "1.8"->8, "9"->9, "10"->10
+  private static final int JAVA_SPEC_VER = Math.max(8, Integer.parseInt(
+      System.getProperty("java.specification.version").split("\\.")[0]));
```

Comment:
JAVA_SPEC_VER is set to 8 even on Java 7. Can we cover Java 7 here? If not, we should have a comment for justification at least.